app/routes/companyInterest/components/CompanyInterestPage.tsx

```diff
@@ -291,18 +142,36 @@ const EventBox = ({
   fields: any;
   language: string;
 }): ReactNode => (
-  <Flex column className={styles.checkboxWrapper}>
-    {fields.map((key, index) => (
-      <Field
-        key={`events[${index}]`}
-        name={`events[${index}].checked`}
-        label={EVENT_TYPES[eventToString(key)][language]}
-        type="checkbox"
-        component={CheckBox.Field}
-        normalize={(v) => !!v}
-      />
-    ))}
-  </Flex>
+  <FormSpy subscription={{ values: true }}>
+    {(props) => {
+      const newFields = fields.map((field) => field);
+      if (props.values.officeInTrondheim !== 'yes') {
+        let indexOfCompanyToCompany = -1;
+        fields.forEach((field, index) => {
+          if (fields.value[index].name == 'company_to_company') {
+            indexOfCompanyToCompany = index;
+          }
+        });
+        if (indexOfCompanyToCompany != -1) {
+          newFields.splice(indexOfCompanyToCompany, 1);
```

Comment:
Is the `indexOfCompanyToCompany` variable necessary? Couldn't we just splice the list inside the forEach-loop?